src/common/Database.js

```diff
@@ -0,0 +1,436 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3"], factory);
+    } else {
+        root.common_Database = factory(root.d3);
+    }
+}(this, function (d3) {
+    function Grid() {
+        this.clear();
+    }
+    Grid.prototype.constructor = Grid;
```

Comment:
this isnt in reference to the layout grid we have ... should we use the term DataMatrix? (this is my first pass through of this code so my questions might not make sense)